homeassistant/components/bthome/sensor.py

```diff
@@ -378,7 +380,9 @@ def sensor_update_to_bluetooth_data_update(
             if description.device_class
         },
         entity_data={
-            device_key_to_bluetooth_entity_key(device_key): sensor_values.native_value
+            device_key_to_bluetooth_entity_key(device_key): cast(
+                float | None, sensor_values.native_value
+            )
```

Comment:
The `cast` type is a result of the annotation on `native_value`.
https://github.com/home-assistant/core/blob/e4a3cab801ded3065c49963e4c777487c74e126b/homeassistant/components/bthome/sensor.py#L419-L420

_`int` is a subtype of `float` and as such redundant._

--
The alternative would be to use the full type `str | float | date | Decimal | None` which is the same as `SensoreEntity.native_value`.